scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala

```diff
@@ -81,13 +53,9 @@ class SoftMax[T: ClassTag](var pos: Int = 1)(implicit ev: TensorNumeric[T])
 }
 object SoftMax{
 
-  def apply[@specialized(Float, Double) T: ClassTag](pos: Int = 1)
-      (implicit ev: TensorNumeric[T]) : SoftMax[T] = {
-    new SoftMax[T](pos)
-  }
-  def apply[@specialized(Float, Double) T: ClassTag]
+  def apply[@specialized(Float, Double) T: ClassTag]()
     (implicit ev: TensorNumeric[T]) : SoftMax[T] = {
-    new SoftMax[T](1)
+    new SoftMax[T]()
   }
   // Notice: SoftMin will call this function
   private[nn] def updateOutput[T: ClassTag](input: Tensor[T], output: Tensor[T],
```

Comment:
are these two methods still used?

Reply:
yes, they are used by Softmin

Reply:
then move them to SoftMin (maybe applying similar optimizations in a separate PR?)

Reply:
Do we need support softmin? Looks like tensorflow and keras doesn't support softmin.

Reply:
We can then keep it as is